draft-ietf-privacypass-key-consistency.md

```diff
@@ -287,105 +299,87 @@ In this model, servers publish keys in an external database and clients fetch ke
 |          |           |
 +----------+           |
                        v
-+----------+         +-+--------+       +----------+
-|          |         |          |       |          |
-|  Client  +-------->+ Database +<------+  Server  |
-|          |         |          |       |          |
-+----------+         +-+--------+       +----------+
-     x                 ^
-     x                 |
++----------+         +-----------+       +----------+
+|          |         |           |       |          |
+|  Client  +-------->+   Cache   +------>+  Server  |
+|          |         |           |       |          |
+|          |         +-----------+       |          |
+|          |                             |          |
+|          |         +-----------+       |          |
+|          |         |           |       |          |
+|          +============ Proxy  ========>+          |
+|          |         |           |       |          |
++----------+         +-+---------+       +----------+
+      x                ^
+      x                |
 +----------+           |
 |          |           |
 |  Client  +-----------+
 |          |
 +----------+
 ~~~
-{: #fig-disc-database title="Database Discovery Example"}
+{: #fig-disc-shared-proxy title="Shared Proxy with Confirmation Discovery Example"}
+
+Ideally, clients confirm with the server via some anonymizing proxy. Examples of proxies
+include anonymous systems such as Tor. Tor proxies are general purpose and operate
+at a lower layer, on arbitrary communication flows, and therefore they are oblivious
+to clients fetching keys. A large set of untrusted proxies that are aware of key fetch
+requests ({{cache-based}}) may be used in a similar way. Depending on how clients
+fetch such keys from servers, it may become more difficult for servers to uniquely
+target individual clients with unique keys without detection. This is especially true
+as the number of clients of these anonymity networks increases. However, beyond
+Tor, there does not exist a special-purpose anonymity network for this purpose.
```

Comment:
I think that this emphasis on Tor as a public utility is a little off.  I'd prefer to see this mention VPNs, proxies, MASQUE, and other things as options, noting that most of these tend to insist on payment for access.

Reply:
Filed #29 to track this.